### PR TITLE
Accelerate 1.8.1 and BNB 0.46.0 update

### PR DESCRIPTION
**Pull request by winglian**
This PR also updates the CI for multigpu e2e to include the dataset_prepared_path set to prefix with the temp_dir so the tests don't block each other from the new FileLock

## Summary by CodeRabbit

* **Chores**
  * Upgraded package versions for `bitsandbytes` and `accelerate` in the requirements.
* **Tests**
  * Added a new dataset preparation path configuration to multiple test suites.
  * Increased validation set sizes and specified training data splits in evaluation tests.
  * Clarified return type annotations for temporary directory fixtures.
* **Bug Fixes**
  * Disabled default PyTorch bfloat16 dtype setting during training due to compatibility issues with certain optimizers.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -15,7 +15,7 @@ huggingface_hub==0.32.2
 peft==0.15.2
 transformers==4.52.4
 tokenizers>=0.21.1
-accelerate==1.7.0
+accelerate==1.8.0
```
Comment (salmanmohammadi) [inline requirements.txt]:
```suggestion
accelerate==1.8.1
```
new accelerate patch just dropped
